compio/src/lib.rs

```diff
@@ -1,141 +1,45 @@
+//! # Compio
+//! A thread-per-core Rust runtime with IOCP/io_uring/polling.
+//! The name comes from "completion-based IO".
+//! This crate is inspired by [monoio](https://github.com/bytedance/monoio/).
 //!
-#![doc = include_str!("../../README.md")]
-#![cfg_attr(feature = "allocator_api", feature(allocator_api))]
-#![cfg_attr(feature = "lazy_cell", feature(lazy_cell))]
-#![cfg_attr(feature = "once_cell_try", feature(once_cell_try))]
-#![cfg_attr(feature = "read_buf", feature(read_buf))]
-#![warn(missing_docs)]
-
-pub mod driver;
-pub mod fs;
-pub mod net;
-pub mod op;
-
-pub use buf::BufResult;
-pub use compio_buf as buf;
+//! ## Quick start
+//! ```rust
+//! # compio::runtime::block_on(async {
+//! use compio::fs::File;
+//!
+//! let file = File::open("Cargo.toml").unwrap();
+//! let (read, buffer) = file
+//!     .read_to_end_at(Vec::with_capacity(1024), 0)
+//!     .await
+//!     .unwrap();
+//! assert_eq!(read, buffer.len());
+//! let buffer = String::from_utf8(buffer).unwrap();
+//! println!("{}", buffer);
+//! # })
+//! ```
 
-#[cfg(target_os = "windows")]
-pub mod named_pipe;
+#![warn(missing_docs)]
 
-#[cfg(feature = "event")]
-pub mod event;
-#[cfg(feature = "runtime")]
-mod key;
-#[cfg(feature = "runtime")]
-pub(crate) use key::Key;
-#[cfg(feature = "runtime")]
-mod attacher;
-#[cfg(feature = "runtime")]
-pub(crate) use attacher::Attacher;
-#[cfg(feature = "signal")]
-pub mod signal;
 #[cfg(feature = "macros")]
+#[doc(no_inline)]
 pub use compio_macros::*;
 #[cfg(feature = "runtime")]
-pub mod task;
+#[doc(no_inline)]
+pub use compio_runtime as runtime;
```

Comment:
I think we should `#[doc(inline)]` these crate-as-module situation like what futures does. It's easier for users to navigate between modules in docs.rs compare to different crates. Also we can pick out some important items like `spawn`.